src/__tests__/test-utils-selectors.test.tsx

```diff
@@ -0,0 +1,141 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import fs from 'fs';
+import Path from 'path';
+import glob from 'glob';
+import { transformSync, types, PluginObj, NodePath } from '@babel/core';
+import { flatten, zip } from 'lodash';
+
+// The test extracts generated selectors from the compiled output and matches those against the snapshot.
+test('test-utils selectors', () => {
+  // Find referenced selector files and properties.
+  const selectorsFilePathToUsedProperties = new Map<string, Set<string>>();
+  for (const file of glob.sync('lib/components/test-utils/selectors/**/*.js')) {
+    extractSelectorProperties(file, (filePath, propertyKey) => {
+      const properties = selectorsFilePathToUsedProperties.get(filePath) ?? new Set();
+      properties.add(propertyKey);
+      selectorsFilePathToUsedProperties.set(filePath, properties);
+    });
+  }
+
+  // Find referenced selector values.
+  const componentToSelectors: Record<string, string[]> = {};
+  for (const [filePath, properties] of selectorsFilePathToUsedProperties) {
+    extractComponentSelectors(filePath, [...properties], selector => {
+      const componentName = getComponentNameFromFilePath(filePath);
+      componentToSelectors[componentName] = [...(componentToSelectors[componentName] ?? []), selector].sort();
+    });
+  }
+  expect(componentToSelectors).toMatchSnapshot();
+});
+
+function extractSelectorProperties(file: string, onExtract: (filePath: string, propertyKey: string) => void) {
+  function extractor(): PluginObj {
+    const selectorVars = new Map<string, string>();
+    return {
+      visitor: {
+        // Find require statements that import selectors.
+        CallExpression(path: NodePath<types.CallExpression>) {
+          if (path.node.callee.type === 'Identifier' && path.node.callee.name === 'require') {
+            const argument = path.node.arguments[0];
+            if (argument.type === 'StringLiteral' && argument.value.endsWith('selectors.js')) {
+              if (path.parent.type === 'VariableDeclarator' && path.parent.id.type === 'Identifier') {
+                selectorVars.set(path.parent.id.name, Path.resolve(file, '..', argument.value));
```

Comment:
Why is there an extra `..`? At least it needs an explaining comment

Reply:
Thanks, I will add a comment